### ISSUE TITLE
pi-hole not answering requests from other subnet

### ISSUE THREAD

**Issue by holysoles** [pi-hole/pi-hole#4670]
### Versions
- Pi-hole version is v5.9.1 (Latest: v5.9.1)
- AdminLTE version is v5.11 (Latest: v5.11)
- FTL version is v5.14 (Latest: v5.14)

### Platform
- OS and version:  Raspbian GNU/Linux 10 (buster) armv7l
- Platform:  Raspberry Pi 4 Model B Rev 1.2 

### Expected behavior

pihole should send resolved IPs back to clients located in a different subnet.

### Actual behavior / bug

pihole receives query, resolves domain to IP, but doesn't return it to clients in the other subnet.

### Steps to reproduce

1. From the pihole web console, view the log live: Tools > Tail pihole.log  
2. From a device in a different subnet (192.168.112.0/24), do a lookup using the pihole IP (is in the 192.168.111.0/24 subnet).
   `dig google.com @192.168.111.2`
3. Back in the pihole web console, observe the request come in.
<img width="568" alt="image" src="https://user-images.githubusercontent.com/31580846/161360548-55208204-5e97-4a9c-b509-afbc2fa21607.png">
4. Observe timeout on client
<img width="711" alt="image" src="https://user-images.githubusercontent.com/31580846/161361306-8555d189-fd24-4aa2-ace5-3ecd065dd513.png">

## Debug Token

- URL:  https://tricorder.pi-hole.net/492EQftj/

### Screenshots

_If applicable, add screenshots to help explain your problem._

### Additional context

I did review the documentation around interface [listening/binding here](https://docs.pi-hole.net/ftldns/interfaces/) and various forum posts, but toggling this setting to the various options showed no change in behavior. This is currently set to "permit all origins"

A packet capture running on my Router showed only traffic from the client->pihole:53, but no response. No drops are shown on my firewall between subnets. Clients in 192.168.112.0/24 are able to resolve IPs using the upstream Windows DNS server (192.168.111.5). All firewall rules that apply to that upstream DNS, are also applied to the pihole IP.

I did try running a re-install of pihole and restoring my config (not just a repair) and there was no change in behavior.

**Comment by yubiuser** [comment]
As you seem to be an advanced user, please try Pi-hole's internal package capture and dump the traffic. You can inspect the pcap file with Wireshark. Are there any replies generated by Pi-hole's embedded `dnsmasq`?

https://docs.pi-hole.net/ftldns/package_dump/

**Comment by holysoles** [comment]
@yubiuser thanks for pointing me in that direction. It does appear a reply is being generated internally
<img width="880" alt="Screen Shot 2022-04-02 at 11 39 22 AM" src="https://user-images.githubusercontent.com/31580846/161392583-60ee8a96-e370-433d-a551-b3483888f2b9.png">

I'll look at doing a capture on the router interface that the pihole is attached to and see if its leaving the device.

**Comment by holysoles** [comment]
From running tcpdump locally on the pihole, listening on eth0 (wired only, wifi is disabled). The device itself is not sending the reply:
<img width="877" alt="Screen Shot 2022-04-02 at 11 57 45 AM" src="https://user-images.githubusercontent.com/31580846/161393294-13892384-e8ca-482b-bcee-5a5ce418b95e.png">

**Comment by yubiuser** [comment]
Mhh... your first screenshot shows that a query response is sent do `192.168.112.5`. Was it sent on another interface then `eth0`?

**Comment by holysoles** [comment]
I ran a tcpdump for all interfaces while doing another query. With a `dns` display filter in wireshark, I still don't see a query response being sent, to anywhere
<img width="884" alt="Screen Shot 2022-04-02 at 4 15 44 PM" src="https://user-images.githubusercontent.com/31580846/161401656-b9ced931-233c-4e68-af86-dc918ced29e4.png">

Interestingly, I happened to be working on forcing DNS requests to my pihole, as some devices these days have hardcoded DNS servers (Google devices, Smart TV, etc). When I set up the below iptables rule for a Source NAT Masquerade, clients in the other subnet now receive DNS query responses. It's unclear to me why this changed the behavior, I'm not very familiar with NAT rules.

```
pkts bytes     target           prot  opt  in         out         source      destination  
18    1282 MASQUERADE  udp  --   any    switch0  anywhere    anywhere   match-set ALL_LAN src udp dpt:domain match-set DNS+DHCP_Servers dst /* NAT-5011 tcp_udp */
```

Some contextual info:
- `switch0` is my LAN switch
- `ALL_LAN` is a network group for `192.168.111.0/16` to include my network's two subnets
- `DHS+DHCP_Servers` is an address group of my pihole and upstream Windows DNS
- In the gui i configured the rule to be for dest port 53, not sure why that doesn't show in the iptables rule.

Anyhow given this is resolved with my router config, I'll close this out.
